### PR TITLE
Certs preview screen (EXPOSUREAPP-13262)

### PR DESCRIPTION
**Pull request by mtwalli**
## Changes
- Overview PDF
- Print PDF 
- Progress message
- Error message 
- Share PDF
## Ticket
- https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-13262
- https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-13329

### CODE REVIEW

File: Corona-Warn-App/src/main/res/values-de/covid_certificate_strings.xml [hunk 1]
```diff
@@ -363,4 +363,16 @@
     <string name="export_all_tooltip_title">Zertifikate exportieren</string>
     <!-- XTXT: Certificate export all tooltip message -->
     <string name="export_all_tooltip_message">In einem Schritt alle Ihre Zertifikate in einem gemeinsamen PDF-Dokument speichern</string>
+    <!-- XTXT: Certificate export all progress title -->
+    <string name="export_all_progress_title">PDF wird erstellt</string>
+    <!-- XTXT: Certificate export all progress message -->
+    <string name="export_all_progress_text">Diese kann einige Minuten dauern.</string>
```
Comment (KathrinLuecke) [inline Corona-Warn-App/src/main/res/values-de/covid_certificate_strings.xml:369]:
```suggestion
    <string name="export_all_progress_text">Dies kann einige Minuten dauern.</string>
```

File: Corona-Warn-App/src/main/res/values-de/covid_certificate_strings.xml [hunk 2]
```diff
@@ -363,4 +363,16 @@
...
+    <!-- XTXT: Certificate export all progress cancel -->
+    <string name="export_all_progress_cancel">Abbrechen</string>
+    <!-- XTXT: Certificate export all error title -->
+    <string name="export_all_error_title">Druckversion anzeigen nicht möglich</string>
+    <!-- XTXT: Certificate export all error message -->
+    <string name="export_all_error_message">Die Druckversion der Zertifikate kann nicht angezeigt werden.</string>
+    <!-- XTXT: Certificate export all error faq button -->
+    <string name="export_all_error_faq">Faq zur Druckversion</string>
```
Comment (KathrinLuecke) [inline Corona-Warn-App/src/main/res/values-de/covid_certificate_strings.xml:377]:
```suggestion
    <string name="export_all_error_faq">FAQ zur Druckversion</string>
```

**Review by KathrinLuecke — CHANGES_REQUESTED** [review]

File: Corona-Warn-App/src/main/java/android/print/FilePrinter.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            printAdapter.onLayout(
+                null,
+                attributes,
+                null,
+                object : LayoutResultCallback() {
+                    override fun onLayoutFinished(info: PrintDocumentInfo, changed: Boolean) {
+                        printAdapter.onWrite(
+                            arrayOf(PageRange.ALL_PAGES),
+                            getOutputFileDescriptor(path, fileName),
+                            CancellationSignal(),
```
Comment (schauersbergern) [inline Corona-Warn-App/src/main/java/android/print/FilePrinter.kt:29]: The App crashes here on my Pixel 6 without calling onWriteFinished or onWriteFailed and without throwing any exception. 
log: ```A/libc: Fatal signal 5 (SIGTRAP), code 1 (TRAP_BRKPT), fault addr 0x78711eeb74 in tid 23259 (DefaultDispatch), pid 23194 (main)```

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/ui/validationresult/common/listitem/AffectedFieldsMapper.kt [hunk 4]
```diff
@@ -86,7 +86,7 @@ private fun certificateValue(field: String, certificate: CwaCovidCertificate): S
         }
 
         is TestCertificate -> when (field) {
-            "t.0.tg" -> certificate.targetName
+            "t.0.tg" -> certificate.targetDisease
```
Comment (schauersbergern) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/ui/validationresult/common/listitem/AffectedFieldsMapper.kt:89]: Why did this property change?

File: Corona-Warn-App/src/main/java/android/print/FilePrinter.kt (hunk 3 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/android/print/FilePrinter.kt:29]: pull the latest changes and try again?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/ui/validationresult/common/listitem/AffectedFieldsMapper.kt (hunk 4 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/ui/validationresult/common/listitem/AffectedFieldsMapper.kt:89]: all certificates have `tg` it was called for TC differently for consistency I renamed here , have a look at VC , RC?

File: Corona-Warn-App/src/main/java/android/print/FilePrinter.kt (hunk 3 above)
Comment (schauersbergern) [reply Corona-Warn-App/src/main/java/android/print/FilePrinter.kt:29]: Perfect, now it works

**Review by schauersbergern — APPROVED** [review]

**Review by chiljamgossow — APPROVED** [review]
works nicely 👍 , generated files are fairly small

**Review by KathrinLuecke — COMMENTED** [review]
UA approved

**Pull request merged**